client/src/components/BookingCard/BookingCard.tsx

```diff
@@ -1,44 +1,81 @@
 import { Typography } from '@mui/material';
-import SettingsIcon from '@mui/icons-material/Settings';
+import RequestStatusButton from '../RequestStatusButton/RequestStatusButton';
 import AvatarDisplay from '../AvatarDisplay/AvatarDisplay';
 import { Box } from '@mui/material';
-import { Booking } from '../../interface/Booking';
+import { Request } from '../../interface/Request';
+import { useState } from 'react';
+import changeRequestStatus from '../../helpers/APICalls/changeRequestStatus';
 
 interface Props {
-  booking: Booking;
+  booking: Request;
   isNextBooking?: boolean;
+  isPastBooking?: boolean;
 }
 
-export default function BookingCard({ booking, isNextBooking }: Props): JSX.Element {
+export default function BookingCard({ booking, isNextBooking, isPastBooking }: Props): JSX.Element {
+  const [status, setStatus] = useState<string>(booking.status);
+
   const date = () => {
-    return (
+    const startTime = booking.startDate.getHours();
+    const endTime = booking.endDate.getHours();
+
+    const formattedDate =
       '' +
-      booking.startTime.getDate() +
+      booking.startDate.getDate() +
       ' ' +
-      booking.startTime.toLocaleString('default', { month: 'long' }) +
+      booking.startDate.toLocaleString('default', { month: 'long' }) +
       ' ' +
-      booking.startTime.getFullYear() +
+      booking.startDate.getFullYear() +
       ', ' +
-      booking.startTime.getHours() +
-      '-' +
-      booking.endTime.getHours() +
-      ' ' +
-      (booking.endTime.getHours() >= 12 ? ' PM' : ' AM')
-    );
+      (startTime > 12 ? startTime - 12 : startTime) +
+      (startTime >= 12 ? 'PM' : 'AM') +
+      ' - ' +
+      (endTime > 12 ? endTime - 12 : endTime) +
+      (endTime >= 12 ? 'PM' : 'AM');
+    return formattedDate;
+  };
+
+  const statusColor = () => {
+    if (status == 'accepted') {
+      return 'rgb(0,0,0)';
+    } else {
+      return 'rgba(0,0,0,0.26)';
+    }
+  };
+
+  const handleStatusChange = (status: string) => {
+    setStatus(status);
+    booking.status = status;
+    changeRequestStatus(booking.id, status);
```

Comment:
Why don't we update the snackbar here based on the success/error response from this API call

Reply:
Good idea.